### PR TITLE
path compare

### PR DESCRIPTION
**Pull request by skv-headless**

### CODE REVIEW

**Comment by TeatroIO** [comment]
I've prepared a stage to preview changes. <a href='http://skv-headless-path-compare.c68bfdc4cac4c32c41e1a922e73062ea.ttrcloud.com' target='_blank'>Open stage</a> or <a href='http://skv-headless-path-compare.c68bfdc4cac4c32c41e1a922e73062ea.ttrcloud.com/teatro.log' target='_blank'>view logs</a>.

**Comment by skv-headless** [comment]
updated #5927 now it much simpler.

**Comment by jvanbaarsen** [comment]
@skv-headless Can you attach an up to date screenshot of this feature? Thanks :)

File: app/services/compare_service.rb [hunk 1]
```diff
@@ -7,12 +7,10 @@ def execute(current_user, source_project, source_branch, target_project, target_
     # Note: Use satellite only when need to compare between two repos
     # because satellites are slower than operations on bare repo
     if target_project == source_project
-      Gitlab::CompareResult.new(
```
Comment (jvanbaarsen) [inline app/services/compare_service.rb]: Is it correct behaviour that you don't use `Gitlab::CompareResult` anymore?

**Comment by skv-headless** [comment]
on 1024px
selection_004

**Comment by jvanbaarsen** [comment]
@skv-headless What happends if no path is given? Is the compare still working without it?

**Comment by skv-headless** [comment]
It will work like before.

**Comment by jvanbaarsen** [comment]
@skv-headless Maybe you can add a placeholder in the text box saying: `Optional - i.e.: example path`

**Comment by skv-headless** [comment]
selection_005
@jvanbaarsen is it ok?

**Comment by sspreitzer** [comment]
+1 I need this feature very much

File: config/routes.rb [hunk 2]
```diff
@@ -257,7 +257,7 @@
       end
 
       get '/compare/:from...:to' => 'compare#show', :as => 'compare',
-          :constraints => { from: /.+/, to: /.+/ }
+          :constraints => { from: /.+/, to: /.+/, path: /.+/  }
```
Comment (jvanbaarsen) [inline config/routes.rb]: @skv-headless Can you please fix this Hound issue?

**Comment by jvanbaarsen** [comment]
@randx What do you think about this overal feature? If you want this to be merged in I'll guide the rest of this PR.

**Comment by randx** [comment]
@jvanbaarsen I like idea of comparing certain directory or file. If it can be done in nice way (UI too) - I would like to have it

**Comment by jvanbaarsen** [comment]
@skv-headless Are you still working on this?

**Comment by skv-headless** [comment]
What should I do? If I understood correct @randx don't like UI, but I don't know how to improve it.

**Comment by sspreitzer** [comment]
@skv-headless: please remove the whitespace after `/.+/` as mentioned from @houndci above.

**Comment by sspreitzer** [comment]
Just add a new commit to the branch used in the pull request and push the branch to GitHub. The pull request will automatically be updated with the additional commit.

**Comment by jvanbaarsen** [comment]
@randx Would the way @skv-headless implemented it now (UI-wise) be acceptable as a first pass? We can improve upon it later on.

File: app/views/projects/compare/show.html.haml [hunk 3]
```diff
@@ -5,6 +5,12 @@
 
 = render "form"
 
+- if compare_to_mr_button?
+  = link_to "Make a merge request »", compare_mr_path,
+    {style: 'font-size: 15px;'}
+  %br
+  %br
+
```
Comment (randx) [inline app/views/projects/compare/show.html.haml]: Please remove this. We already have `new mr` button on this page

**Comment by randx** [comment]
> We can improve upon it later on.

@jvanbaarsen my experience says it would be me :smile: . So I rather finish UI in this PR. Try to hide path text field and expand it only when needed. Same like we did for import URL on new project page

**Comment by jvanbaarsen** [comment]
@randx Well, yeah.. it would probably be you :trollface: Ok. I like what you're proposing. @skv-headless What do you think?

**Comment by jvanbaarsen** [comment]
This merge request has been closed because a request for more information has not been reacted to for more than 2 weeks. If you respond and conform to the merge request guidelines in our [contributing guidelines](https://gitlab.com/gitlab-org/gitlab-ce/blob/master/CONTRIBUTING.md#pull-requests) we will reopen this merge request.

**Comment by sspreitzer** [comment]
:,(

**Comment by cruelbob** [comment]
@skv-headless Are you going to continue working on this task? This is verty useful feature!

**Comment by dhoffend** [comment]
I'm missing this feature as well

**Comment by rutsky** [comment]
Ough, this is a killer feature for me, too bad it's abandoned!

I have [huge PR](https://github.com/buildbot/buildbot/pull/1962) (300+ changed files), and to get it reviewed I want to provide separate document with high level description of changes with links on some details in overall PR diff (i.e. links on diffs for specific files).
And nobody can show my changes in per-file mode! I tried GitHub, BitBucket, Phabricator's Differential, Reviewable, Review Board.

**Comment by erykciepiela** [comment]
We would also appreciate such feature. Is there any chance someone is working on this currently or we're waiting for a volunteer?

**Comment by charlespockert** [comment]
Definitely would like this - diffing 1 file gives you a history of the file rather than a history of the full commits and is *so* useful!

**Pull request closed without merging**